install.bat

```diff
@@ -471,6 +471,47 @@ local function get_visual_studio_directory()
 	return nil
 end
 
+local function get_visual_studio_directory_from_vswhere()
+	assert(type(vars.LUA_RUNTIME)=="string", "requires vars.LUA_RUNTIME to be set before calling this function.")
+	local major, minor = vars.LUA_RUNTIME:match('VCR%u*(%d+)(%d)$')
+	if not major then
+		print(S[[    Cannot auto-detect Visual Studio version from $LUA_RUNTIME]])
+		return nil
+	end
+	if tonumber(major) < 14 then
+		return nil

```

Comment:
Should these code paths have error messages?

Reply:
Not sure but I prefer not. Every `return nil` could just be a normal case. For example `tonumber(major) < 14` could be true if user is using VS lower than 2015. `program_dir` could be empty if the system architecture is x86. `vswhere` could be not found if using VS2015 or an early version of VS2017. These are regular cases that luarocks might just want to fallback to other means of finding a VS installation.

Reply:
@daurnimator Do you suggest to keep the current code as-is, or we add additional error messages (that may bother pre-VS2017 user on Windows)?